backend/src/middleware/notifications/notificationsMiddleware.js

```diff
@@ -93,29 +97,66 @@ const notifyUsersOfComment = async (label, commentId, postAuthorId, reason, cont
     await session.writeTransaction(async transaction => {
       await transaction.run(
         `
-        MATCH (postAuthor:User {id: $postAuthorId})-[:WROTE]->(post:Post)<-[:COMMENTS]-(comment:Comment { id: $commentId })<-[:WROTE]-(commenter:User)
-        WHERE NOT (postAuthor)-[:BLOCKED]-(commenter)
-        MERGE (comment)-[notification:NOTIFIED {reason: $reason}]->(postAuthor)
-        SET notification.read = FALSE
-        SET (
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-20T18:01:19Z" title="Monday, January 20th 2020, 7:01:19 pm +01:00">Jan 20, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -93,29 +96,66 @@ const notifyUsersOfComment = async (label, commentId, postAuthorId, reason, cont
     await session.writeTransaction(async transaction => {
       await transaction.run(
         `
-        MATCH (postAuthor:User {id: $postAuthorId})-[:WROTE]->(post:Post)<-[:COMMENTS]-(comment:Comment { id: $commentId })<-[:WROTE]-(commenter:User)
-        WHERE NOT (postAuthor)-[:BLOCKED]-(commenter)
-        MERGE (comment)-[notification:NOTIFIED {reason: $reason}]->(postAuthor)
-        SET notification.read = FALSE
-        SET (
-        CASE
-        WHEN notification.createdAt IS NULL
-        THEN notification END ).createdAt = toString(datetime())
-        SET notification.updatedAt = toString(datetime())
-      `,
+          MATCH (postAuthor:User {id: $postAuthorId})-[:WROTE]->(post:Post)<-[:COMMENTS]-(comment:Comment { id: $commentId })<-[:WROTE]-(commenter:User)
+          WHERE NOT (postAuthor)-[:BLOCKED]-(commenter)
+          MERGE (comment)-[notification:NOTIFIED {reason: $reason}]->(postAuthor)
+          SET notification.read = FALSE
+          SET ( CASE WHEN notification.createdAt IS NULL THEN notification END ).createdAt = toString(datetime())
```


Can't you use `COAELESCE`? I find this quite hard to read.
https://neo4j.com/docs/cypher-manual/current/functions/scalar/#functions-coalesce

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-02-05T12:00:37Z" title="Wednesday, February 5th 2020, 1:00:37 pm +01:00">Feb 5, 2020</time>_

---


COOL stuff! 😍
Thanks for your hint @roschaefer !